inst/include/plant/species.h

```diff
@@ -192,6 +195,45 @@ std::vector<double> Species<T,E>::net_reproduction_ratio_by_cohort() const {
   return ret;
 }
 
+template <typename T, typename E>
+std::vector<double> Species<T,E>::integrate_shared_resource_consumption(const E& environment) const {
+  int n_resources = environment.vars.state_size;
+  std::vector<double> ret;
+  ret.reserve(n_resources);
+
+  for(size_t i = 0; i < n_resources; i++){
+    std::vector<double> heights;
```

Comment:
It's inefficient to get cohort heights within the this loop because they're the constant for all resources (soil layers). I'll make a suggestion elsewhere how we can remove the loop over `n_resources` altogether.